src/server/lib/okta/error.ts

```diff
@@ -0,0 +1,14 @@
+// https://developer.okta.com/docs/reference/error-codes/
+export interface OktaError {
+  errorCode: string;
+  errorSummary: string;
+  errorLink: string;
+  errorId: string;
+  errorCauses: [string?];
+}
+
+export interface FetchOktaError {
+  status: number;
+  statusText: string;
+  oktaError: OktaError;
+}
```

Comment:
> This is really great work!
> 
> I got most of the way through but then got stuck around the OIDC stuff. Maybe we can pair on that?
> 
> One general question: Is there a plan to build the sign in flow against idapi? If so, will this PR conflict with that or can we switch back?

Yeah sure, let's pair on this. I might take the engineers through it as a team too, since it would be useful to get multiple eyes on this. Discuss at standup.